content/components/banner.mdx

```diff
@@ -8,15 +8,256 @@ tags:
+  </Box>
+</Box>
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="upsell banner"
+    src="https://github.com/primer/design/assets/813754/48b37115-c0e0-40c3-83f1-3e831178f175"
+  />
+  <Box as="p" mt="0">
+    <h4>Upsell</h4>
+    Use upsell banners to inform the users about a feature that can be enabled by upgrading their plan.
+  </Box>
+</Box>
+
+### Title
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="banner without title and banner with title"
+    src="https://github.com/primer/design/assets/813754/c5876803-7b5b-486d-b2ec-27fcfdf0c5d8"
+  />
+  <Box as="p" mt="0">
+    <h4>Title</h4>
+    Use a title to consisly present the topic of the banner when the message of the banner is complex and too long to scan.
+  </Box>
+</Box>
+
+### Dismissable
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="banners with a dismiss button"
+    src="https://github.com/primer/design/assets/813754/e9863c48-cf3a-472b-a066-0ba139a633ca"
+  />
+  <Box as="p" mt="0">
+    <h4>Dismiss button</h4>
+    Banners with an <i>onDismiss</i> property automatically show a dismiss button.
+  </Box>
+</Box>
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="a critical banner with no actions and one with a single action"
+    src="https://github.com/primer/design/assets/813754/2463e14c-5cb9-4bb4-9e3a-474c60c32da4"
+  />
+  <Box as="p" mt="0">
+    <h4>Critical banner</h4>
+    <p>
+    Critical banner can't be made dismissable. Since they should only be used for errors that need to be resolved, the only way to get rid of a critical banner is to resolve the error. If they are used to warn the user about destructive actions the banner should also not be dismissable.</p>
+    <p>
+    If the error must be resolved on a different page, it is recommended to add an action that directs the user to the page where the error can be resolved.
+    </p>
+  </Box>
+</Box>
+
+### Actions
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="trailing actions on a banner"
+    src="https://github.com/primer/design/assets/813754/b4d2e88e-2f0e-4b29-8f21-13ac8399c943"
+  />
+  <Box as="p" mt="0">
+    <h4>Actions</h4>
+    A maximum of 2 actions can be added to a banner. Whenever possible it is recommended to only add a single action to better direct the users attention.
```

Comment:
The text after the first paragraph is being rendered as a code block. This can be resolved by following my MDX formatting suggestion in the inline message guidelines PR: https://github.com/primer/design/pull/746#discussion_r1536118986

<img width="842" alt="Screenshot 2024-03-22 at 4 05 20 PM" src="https://github.com/primer/design/assets/2313998/bdc54012-be59-4fe0-9c61-0311e66759ad">